core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/YugabyteDefaultLoadBalancingPolicy.java

```diff
@@ -0,0 +1,271 @@
+
+    // For YCQL, when localDC is provided, use the DefaultNodeFilterHelper to find
+    // out the local nodes and also maintain list of all the live nodes in every DC.
+    if (!StringUtils.isBlank(localDc)) {
+
+      if (filter.test(node)) {
+        distanceReporter.setDistance(node, NodeDistance.LOCAL);
+        if (node.getState() != NodeState.DOWN) {
+          // This includes state == UNKNOWN. If the node turns out to be unreachable, this
+          // will be
+          // detected when we try to open a pool to it, it will get marked down and this
+          // will be
+          // signaled back to this policy
+          liveNodesInLocalDc.add(node);
+          liveNodesInAllDC.add(node);
+        }
+      } else {
+
+        // For Multi-DC/Multi-region YugabyteDB Clusters, leader tablets can be present
+        // in the DC's
+        // other than the DC considered as LOCAL. Hence we'll need to consider the nodes
+        // from
+        // other DC's as live for routing the YCQL operations.
+        if (node.getState() == NodeState.UP || node.getState() == NodeState.UNKNOWN) {
+          liveNodes.add(node);
+          liveNodesInAllDC.add(node);
+          LOG.debug(
+              "[{}] Adding {} as it belongs to the {} DC in Multi-DC/Region Setup",
+              logPrefix,
+              node,
+              node.getDatacenter());
+          distanceReporter.setDistance(node, NodeDistance.REMOTE);
+        } else {
+          distanceReporter.setDistance(node, NodeDistance.IGNORED);
+        }
+      }
+    } else {
+      // For YCQL, When Local DC is not provided, all the available UP nodes will be
+      // considered
+      // local
+      distanceReporter.setDistance(node, NodeDistance.LOCAL);
+      if (node.getState() != NodeState.DOWN) {
+        // This includes state == UNKNOWN. If the node turns out to be unreachable, this
+        // will be
+        // detected when we try to open a pool to it, it will get marked down and this
+        // will be
+        // signaled back to this policy
+        liveNodesInAllDC.add(node);
+      }
+    }
+  }
+  
+  private boolean handleNodeDownEvent(Node node) {
+	  boolean handleSuccess = false;
+	  
+	  if (liveNodesInAllDC.contains(node)) {
+		  liveNodesInAllDC.remove(node);
+		  handleSuccess = true;
+	  }
+	  
+	  if (liveNodesInLocalDc.contains(node)) {
+		  liveNodesInLocalDc.remove(node);
+		  handleSuccess = true;
+	  }
+	  
+	  if (liveNodes.contains(node)) {
+		  liveNodes.remove(node);
+		  handleSuccess = true;
+	  }
+	  
+	  return handleSuccess;
+  }
+
+  private ConsistencyLevel findConsistencyLevelForRequest(Request request) {
+
+    // By default, All YCQL statements will have YB_STRONG consistency level
+    ConsistencyLevel statementConsistencyLevel = ConsistencyLevel.YB_STRONG;
+
+    if (request instanceof Statement) {
+
+      Statement<?> ycqlStatement = (Statement<?>) request;
+
+      statementConsistencyLevel =
```

Comment:
nit:

Since the default is already assigned above and since we are checking the ycqlStatement's consistency level against null then just assign only when it is non-null. else unnecessary again assigning the default.

Reply:
Review comment incorported.